crates/typst-library/src/foundations/symbol.rs

```diff
@@ -198,12 +198,12 @@ impl Symbol {
         #[variadic]
         variants: Vec<Spanned<SymbolVariant>>,
     ) -> SourceResult<Symbol> {
-        let mut list = Vec::new();
+        let mut list = Vec::<(EcoString, _)>::new();
         if variants.is_empty() {
             bail!(span, "expected at least one variant");
         }
         for Spanned { v, span } in variants {
-            if list.iter().any(|(prev, _)| &v.0 == prev) {
+            if list.iter().any(|(prev, _)| same_modifiers(&v.0, prev)) {
                 bail!(span, "duplicate variant");
```

Comment:
Maybe we should improve the error message (and/or hints) to point out which variant is duplicate? And also include a hint, if `&v.0 != prev` (we can check this by replacing `.any` with `.find`), that the order of variants doesn't matter, so people who were confused like me don't remain so afterwards.

Reply:
The error message is already at the location of the duplicated variant, so specifying which variant is duplicated is redundant. I agree a hint should be shown when the modifiers are specified in different orders.

Reply:
This is a bit of a common theme. We've generally landed on always specifying it redundantly since the argument could always be a variable. It also makes the error message a bit more useful standalone.

Reply:
That makes sense

Reply:
maybe "variants with the same modifiers are the same, regardless of their order or amount"?